core/src/main/scala/cats/Foldable.scala

```diff
@@ -266,6 +266,12 @@ import simulacrum.typeclass
     !isEmpty(fa)
 
   /**
+    * Find the first element that when mapped by `f` yields a non-empty Option, returning the mapped value. Otherwise return `None`.
+    */
+  def findMap[A, B](fa: F[A])(f: A => Option[B]): Option[B] =
+    foldRight(fa, Eval.now(Option.empty[B]))((x, ly) => Eval.now(f(x).orElse(ly.value))).value
```

Comment:
I think a better way to do this would be:

```scala
def findMap[A, B](fa: F[A])(f: A => Option[B]): Option[B] =
  foldRight(fa, Eval.now(Option.empty[B])) { (x, ly) =>
    f(x) match {
      case None => ly
      case o => Eval.now(o)
    }
  }
```

This probably means I didn't do a good job of documenting `Eval`. The basic issue here is that you generally don't want to see a call to `.value` inside an `Eval` constructor (e.g. `.now`) if you can help it, especially a recursive call like the function call in `.foldRight`. The reason is that this ends up defeating the trampolining that `Eval` can do (and is a reason that some people objected to providing the `.value` method at all).

One way to see if your implementation has this issue is to construct a very large list and then search for the last element, e.g. search `(0 to 20000).toList` for `20000`.